File: src/main/java/com/example/roombooking/models/NotificationsTable.java
package com.example.roombooking.models;

import lombok.Data;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.JoinColumn;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "notification")
public class NotificationsTable {
    @Id
    private int notificationid; 
    private int userid; 
    private String message;
    private boolean isread; 
    private LocalDateTime createdat;

    @ManyToOne
    @JoinColumn(name = "userid", insertable = false, updatable = false)
    private UsersTable user;
}

File: Frontend - Kotlin/app/src/main/java/com/example/project/models/NotificationsTable.kt
package com.example.project.models

import com.google.gson.annotations.SerializedName

data class NotificationsTable(
    @SerializedName("notificationid") val notificationId: Int,
    @SerializedName("userid") val userId: Int,
    @SerializedName("message") val message: String,
    @SerializedName("isread") val isRead: Boolean,
    @SerializedName("createdat") val createdAt: String
)